platform/platform-filter-delegate/src/main/java/org/codice/ddf/platform/filter/delegate/DelegateServletFilter.java

```diff
@@ -14,104 +14,286 @@
+                            securityFilterServiceReference,
+                            securityFilter);
                 }
-                iterator.remove();
+
+                chain.addSecurityFilter(securityFilter);
             }
 
             chain.doFilter(servletRequest, servletResponse);
         } else {
-            LOGGER.debug("Did not find any filters");
+            LOGGER.debug("Did not find any SecurityFilters. Acting as a pass-through filter...");
             filterChain.doFilter(servletRequest, servletResponse);
         }
     }
 
-    @Override
-    public void init(FilterConfig filterConfig) throws ServletException {
-        this.filterConfig = filterConfig;
+    private boolean hasBeenInitialized(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference,
+            final BundleContext bundleContext) {
+        return keysOfInitializedSecurityFilters.contains(getFilterKey(securityFilterServiceReference,
+                bundleContext));
+    }
+
+    private void initializeSecurityFilter(BundleContext bundleContext,
+            ServiceReference<SecurityFilter> securityFilterServiceReference,
+            SecurityFilter securityFilter) throws ServletException {
+        final ServletContext servletContext;
+        final String filterName = getFilterName(securityFilterServiceReference, bundleContext);
+
+        if (filterConfig == null) {
+            LOGGER.warn(
+                    "DelegateServletFilter was initialized with a null filterConfig. Initializing SecurityFilter {} with null ServletContext",
+                    filterName);
+            servletContext = null;
+        } else {
+            servletContext = filterConfig.getServletContext();
+        }
+
+        securityFilter.init(new SecurityFilterInitFilterConfig(securityFilterServiceReference,
+                servletContext,
+                bundleContext));
+        keysOfInitializedSecurityFilters.add(getFilterKey(securityFilterServiceReference,
+                bundleContext));
+        LOGGER.debug("Initialized SecurityFilter {}", filterName);
+    }
+
+    public void removeSecurityFilter(
+            final ServiceReference<SecurityFilter> securityFilterServiceReference) {
+        final BundleContext bundleContext = getContext();
+
+        if (bundleContext != null) {
+            // unmark the SecurityFilter as initialized so that it can be re-initialized if the SecurityFilter is registered again
+            keysOfInitializedSecurityFilters.remove(getFilterKey(securityFilterServiceReference,
+                    bundleContext));
+            bundleContext.getService(securityFilterServiceReference)
+                    .destroy();
+        } else {
+            LOGGER.warn(
+                    "Unable to remove SecurityFilter {}. Try restarting the system or turning up logging to monitor current SecurityFilters.");
+        }
+    }
+
+    @Nonnull
+    private static String getFilterKey(
```

Comment:
I was trying to be clear that the filter name is what it being used as the unique identifier for each `SecurityFilter`. This method could be refactored to use something else for the key, and now it's clear exactly what places in the code want the key (which is used for `keysOfInitializedSecurityFilters`) and what places actually want the filter name.